Update target data format and file structure organization documentation (#243)
hubverse-org/hubDocs: docs/source/user-guide/hub-structure.md

2. If code is included in the hub repository, it should live in a centrally located directory, which we recommend naming `src`.
3. If code has the potential to disrupt or break other continuous integration operations in the hub (e.g., validation of incoming submissions),
it should be moved to another repository.

> 4. Large target data files may be partitioned, but they should be stored in parquet format in the `target-data` directory and follow Apache Hive naming conventions (see Data and Code section below). 

Reviewer: ```suggestion
4. Target data files must be stored in the `target-data` directory. Large target data files may be partitioned, but they should be stored in parquet format and follow Apache Hive naming conventions (see Data and Code section below). 
```

| Component | Location | Description | Hub provides | Modeler provides |
| ------ | ------ | ------ | ------ | ------ |
| Target data directory | `target-data/` | Folder storing actual observed (i.e., target) values of an outcome (or links to external open-access sources) and information on how  model targets can be calculated from target data | X | |

> | Time series data | `target-data/time-series.csv` or .parquet | File with observed counts or rates partitioned for each unique combination of `task id` values | X |  | 
> | Oracle output data | `target-data/oracle-output.csv` or .parquet | File containing data derived from the time series data; represents the model output that would have been generated if the target data values were known ahead of time. For parquet files, column data types in the `oracle-output` file(s) should be consistent with the schema defined in the config file, with the `oracle-value` column having the same data type as the model output `value` column. | X |  |

Reviewer: ```suggestion
| Time series data | `target-data/time-series.csv` or `.parquet` | File with observed counts or rates partitioned for each unique combination of `task id` values | X |  | 
| Oracle output data | `target-data/oracle-output.csv` or `.parquet` | File containing data derived from the time series data; represents the model output that would have been generated if the target data values were known ahead of time. For parquet files, column data types in the `oracle-output` file(s) should be consistent with the schema defined in the config file, with the `oracle-value` column having the same data type as the model output `value` column. | X |  |
```

| Auxiliary data directory | `auxiliary-data/` | Folder storing any additional data related to modeling efforts | X |  |
| Source code directory | `src/` | Folder storing code that is present in the hub repository, including code to access target time series data and/or oracle output programmatically | X |  |
```

> * Partitioned target data files should be stored in the `target-data` directory, in either a `target-data/times-series` or `target-data/oracle-output` subdirectory.  
> * Partitioned target data should follow Apache Hive naming conventions. In Apache Hive, the file name format of partitioned data depends on the partition column names and their values. The files corresponding to each partition are stored in subdirectories, and the directory names encode the partition column names and their values, e.g. <partition_column_1>=<value_1>/<partition_column_2>=<value_2>/.../<data_files>. This means Hive-style partitioned data subdirectories are self describing and can be easily read by partition-aware data readers.
> 
> * Here's an example of oracle output data in the `target-data/oracle-output` directory partitioned by target_end_date:
> 
> ```{code block} json
> ├── target_end_date=2023-06-03
> │   └── part-0.parquet
> ├── target_end_date=2023-06-10
> │   └── part-0.parquet
> └── target_end_date=2023-06-17
>     └── part-0.parquet
> ```
> 
> * Hubs can use their own file naming convention or retain the file names that are generated by the library they use for partitioning. Since Hive-partitioned datasets are not expected to store data in the file names themselves, tooling to read them ignores file names altogether. 

Reviewer: Placing this in an admonition block makes it clear that it's additional info for time series data partitions 

![Screen Shot showing box with blue border that says "Partitioning Target Data" with contents from the text below](https://github.com/user-attachments/assets/69496e6e-caa7-4928-a6c7-5271df711f9a)

````suggestion

:::{admonition} Partitioning Target Data
:class: note

* Partitioned target data files should be stored in the `target-data` directory, in either a `target-data/times-series` or `target-data/oracle-output` subdirectory.
* Partitioned target data should follow Apache Hive naming conventions. In Apache Hive, the file name format of partitioned data depends on the partition column names and their values. The files corresponding to each partition are stored in subdirectories, and the directory names encode the partition column names and their values, e.g. `<partition_column_1>=<value_1>/<partition_column_2>=<value_2>/.../<data_files>`. This means Hive-style partitioned data subdirectories are self describing and can be easily read by partition-aware data readers.

* Here's an example of oracle output data in the `target-data/oracle-output` directory partitioned by `target_end_date`:

    ```
    ├── target_end_date=2023-06-03
    │   └── part-0.parquet
    ├── target_end_date=2023-06-10
    │   └── part-0.parquet
    └── target_end_date=2023-06-17
        └── part-0.parquet
    ```

* Hubs can use their own file naming convention or retain the file names that are generated by the library they use for partitioning. Since Hive-partitioned datasets are not expected to store data in the file names themselves, tooling to read them ignores file names altogether.

:::

````
